requests-for-comments/2022-09-08-notifications.md

```diff
@@ -0,0 +1,81 @@
+# Request for comments: Notifications & @pauldambra 
+
+## Problem statement
+*Who are we building for, what are their needs (include example cases), why is this important?*
+
+Users don't know when to come back to PostHog. Something important can happen and they might miss it. This can be important things in the data, or that another user has edited your insight.
+
+## Success criteria
+*How do we know if this is successful (i.e. metrics, customer feedback), what's out of scope, whats makes this ambitious?*
+
+We have users from [the August 2022 customer interviews](https://github.com/PostHog/posthog/issues/11368) that we can revisit.
+
+We can measure:
+
+* number of triggers created
+* number of notifications sent
+* stickiness of dashboards and insights
+
+## Context
+*What are our competitors doing, what are the technical constraints, what are customers asking for, what does the data tell us, are there external motivations (e.g. launch week, enterprise contract)?*
+
+## Design 
+*What are the key user experience and technical design decisions / trade-offs?*
```

Comment:
RFC Feedback: I'm having a hard time parsing this (the diagram makes some sense after staring at it for a while, but lacks context like why this, what other alternatives we considered, and how it solves the notification problem)

Reply:
Would be also nice to read about what challenges (technical & design wise) this solution might pose, that we're explicitly now accepting.

Reply:
The challenge with notifications are:
- Scope - Users belong to organizations, organizations can have multiple projects, users can be at any level in the hierarchy at any given time - what's the appropriate scope of notifications based on what we know about the user? If the user is in a project, would we notify them of events from another project? Only a subset of users might get notifications about billing or org management level changes. 
- Establishing what all justifies a notification. There's a really broad area to cover with our product, not all things will be of equal importance to a user, not all things are available depending on your plan / payment status / org role / etc. 
- Nailing the preferences - getting just enough configuration to cover the use cases discovered in interviews. This can be a really complex thing to figure out.
- Probably some other stuff... I will keep thinking about it.

Reply:
> RFC Feedback: I'm having a hard time parsing this (the diagram makes some sense after staring at it for a while, but lacks context like why this, what other alternatives we considered, and how it solves the notification problem)

I'll make another pass later. Hard drawing alone :)